client/application/api/api.factory.js

```diff
@@ -14,7 +14,9 @@
       login: login,
       logout: logout,
       getRoomList: getRoomList,
-      reserveSlot: reserveSlot
+      reserveSlot: reserveSlot,
```

Comment:
You know how Python is duck-typed? And that the IDE can't always match parameters in method calls unless they're virtually identical to the method signature itself? Well I would rename this thing `reserveSlot` to just `reservation` since I think it is a reservation object?